### PR TITLE
getUserSettings and setAccepterDisclaimer RPCs

### PR DESCRIPTION
**Pull request by buoyad**
r? @zapu @patrickxb

### CODE REVIEW

**Comment by zapu** [comment]
I'll do the review, thanks

File: go/stellar/remote/remote.go [hunk 1]
```diff
@@ -437,3 +437,32 @@ func SetAccountDefaultCurrency(ctx context.Context, g *libkb.GlobalContext, acco
...
+func GetUserSettings(ctx context.Context, g *libkb.GlobalContext) (stellar1.UserSettings, error) {
+	apiArg := libkb.APIArg{
+		Endpoint:    "stellar/disclaimer",
+		SessionType: libkb.APISessionTypeREQUIRED,
+		NetContext:  ctx,
+	}
+	var apiRes disclaimerResult
+	err := g.API.GetDecode(apiArg, &apiRes)
+	if err != nil {
+		return stellar1.UserSettings{AcceptedDisclaimer: false}, err
```
Comment (zapu) [inline go/stellar/remote/remote.go]: just do
`return stellar1.UserSettings{}, err`
or better:
- change function signature to
`func GetUserSettings(ctx context.Context, g *libkb.GlobalContext) (res stellar1.UserSettings, err error) {`
- change this return to `return res, err`

File: go/stellar/stellarsvc/frontend.go [hunk 2]
```diff
@@ -192,6 +192,33 @@ func (s *Server) GetDisplayCurrenciesLocal(ctx context.Context, sessionID int) (
 	return currencies, nil
 }
 
+func (s *Server) GetUserSettingsLocal(ctx context.Context, sessionID int) (userSettings stellar1.UserSettings, err error) {
+	ctx = s.logTag(ctx)
+	defer s.G().CTraceTimed(ctx, "GetUserSettingsLocal", func() error { return err })()
+	err = s.assertLoggedIn(ctx)
+	if err != nil {
+		return stellar1.UserSettings{}, err
```
Comment (zapu) [inline go/stellar/stellarsvc/frontend.go]: same here, can just `return userSettings, err`, so you don't have to repeat the return type

File: go/stellar/stellarsvc/frontend.go [hunk 3]
```diff
@@ -192,6 +192,33 @@ func (s *Server) GetDisplayCurrenciesLocal(ctx context.Context, sessionID int) (
...
+func (s *Server) SetAcceptedDisclaimerLocal(ctx context.Context, sessionID int) (err error) {
+	ctx = s.logTag(ctx)
+	defer s.G().CTraceTimed(ctx, "SetAcceptedDisclaimerLocal", func() error { return err })()
+	err = s.assertLoggedIn(ctx)
+	if err != nil {
+		return err
+	}
+
+	err = remote.SetAcceptedDisclaimer(ctx, s.G())
+	return err
```
Comment (zapu) [inline go/stellar/stellarsvc/frontend.go]: `return remote.SetAcceptedDisclaimer(ctx, s.G())`

File: protocol/avdl/stellar1/local.avdl [hunk 4]
```diff
@@ -50,6 +50,12 @@ protocol local {
   // supported currencies, returned from getDisplayCurrenciesLocal.
   void changeDisplayCurrencyLocal(int sessionID, AccountID accountID, OutsideCurrencyCode currency);
 
+  record UserSettings {
+    boolean acceptedDisclaimer; // whether the user has accepted the usage disclaimer
+  }
+  UserSettings getUserSettingsLocal(int sessionID);
+  void setAcceptedDisclaimerLocal(int sessionID);
```
Comment (zapu) [inline protocol/avdl/stellar1/local.avdl]: I don't see these RPCs in the spec. Is the `get` one going to be superseded by `GetWalletSettings` ?

**Comment by buoyad** [comment]
@zapu thanks for the comments! changed to use named return vars. PTAL

File: protocol/avdl/stellar1/local.avdl (hunk 4 above)
Comment (buoyad) [reply protocol/avdl/stellar1/local.avdl]: The names are confusing to me right now, if `Wallet` will eventually refer to the feature as a whole (and`Account` to what is now called a `Wallet`), then yes it'll become `getWalletSettings`.

**Comment by buoyad** [comment]
@zapu sounds good to me!

**Pull request merged**